deepali/losses/functional.py

```diff
@@ -638,6 +640,60 @@ def ssd_loss(
     return loss
 
 
+def sample_index_from_mask(mask: Tensor, num_samples: int):
+    r""" Randomly sample locations within a region of interest mask, using the mask as multinomial distribution"""
+    num_locations = mask.size()[2]
+    if num_locations > 2 ** 24:
+        # As torch.multinomial() works only for at most 2^24 categories, use inverse transform sampling
+        # if the number of candidate voxels is very large.
+        mask_cdf = mask.squeeze(1)
+        mask_cdf = torch.cumsum(mask_cdf / mask_cdf.sum(dim=1, keepdim=True), dim=1, out=mask_cdf)
+        mask_cdf = torch.divide(mask_cdf, mask_cdf[:, -1].unsqueeze(-1), out=mask_cdf)
+        cdf_values = torch.rand(mask.size()[0], num_samples).type_as(mask)
+        sample_index = torch.searchsorted(mask_cdf, cdf_values)
+        sample_index = torch.clip(sample_index, 0, num_locations - 1, out=sample_index)
+    else:
+        sample_index = torch.multinomial(mask.squeeze(1), num_samples, replacement=False)
+    return sample_index
+
+
+def sample_in_mask(
```

Comment:
What do you think about naming this function `rand_sample` and move it to `deepali.core.image`?

The function could be modified to take in one or more tensors to sample from with the same random set of indices.

```
def rand_sample(data: Union[Tensor, Tuple[Tensor, ...]], mask: Optional[Tensor] = None, ...) -> Union[Tensor, Tuple[Tensor, ...]]:
    ...

```